apps/extension/src/tvm_ext.cc

```diff
@@ -19,12 +22,57 @@ template<>
 struct extension_class_info<tvm_ext::IntVector> {
   static const int code = 17;
 };
+template<>
+struct array_type_index<tvm_ext::NDSubClass> {
+  static const int code = 1;
+};
 }  // namespace tvm
 }  // namespace runtime
 
 using namespace tvm;
 using namespace tvm::runtime;
 
+namespace tvm_ext {
+class NDSubClass : public tvm::runtime::NDArray {
+ public:
+  class SubContainer : public NDArray::Container {
+   public:
+    SubContainer(bool is_tracing) {
+      array_type_index_ = array_type_index<NDSubClass>::code;
+      is_tracing_ = is_tracing;
+    }
+    static bool Is(NDArray::Container *container) {
+      SubContainer *c = static_cast<SubContainer*>(container);
+      return c->array_type_index_ == array_type_index<NDSubClass>::code;
+    }
+    bool is_tracing_{false};
```

Comment:
simplify the example a bit, by recording a simple integer field like addtional_info.

Reply:
Will do